powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/fido2/PowerAuthAssertionProvider.java

```diff
@@ -173,9 +176,12 @@ public AssertionChallenge failAssertion(String challengeValue, AuthenticatorDeta
             operationFailApprovalRequest.setOperationId(operationId);
             operationFailApprovalRequest.getAdditionalData().putAll(prepareAdditionalData(authenticatorDetail, authenticatorData, clientDataJSON));
 
-            final ActivationRecordEntity activationWithLock = repositoryCatalogue.getActivationRepository().findActivationWithLock(authenticatorDetail.getActivationId());
+            final ActivationRecordEntity activation = activationQueryService.findActivationForUpdate(authenticatorDetail.getActivationId()).orElseThrow(() -> {
+                logger.info("Activation not found, activation ID: {}", authenticatorDetail.getActivationId());
+                return new Fido2AuthenticationFailedException("Activation not found");
```

Comment:
I suggest to keep the context in the exception.
```suggestion
                return new Fido2AuthenticationFailedException("Activation ID: %s not found".formatted(authenticatorDetail.getActivationId()));
```